code/modules/antagonists/heretic/heretic_focus.dm

```diff
@@ -11,6 +11,14 @@
 	RegisterSignal(target, COMSIG_ITEM_EQUIPPED, .proc/on_equip)
 	RegisterSignal(target, COMSIG_ITEM_DROPPED, .proc/on_drop)
 
+	var/obj/item/item_target = target
+	// If our loc is a living mob, it's possible we already have it equippied
+	if(isliving(item_target.loc))
+		var/mob/living/wearer = item_target.loc
+		// We're not being held, that means we're PROBABLY equipped. Give the trait out.
+		if(!(target in wearer.held_items))
+			ADD_TRAIT(item_target.loc, TRAIT_ALLOW_HERETIC_CASTING, ELEMENT_TRAIT(target))
```

Comment:
what if you check the mob's get_item_by_slot(item_target.slot_flags) instead? Though this could backfire if slot_flags has multiple values.

Problem with this current solution is that if the item fits in the pockets, then it'll count as 'equipped', not sure if that's your intention.